modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/toast/commands/OpenToast.es.js

```diff
@@ -51,16 +58,27 @@ const getRootElement = ({container, containerId}) => {
 		}
 	}
 
-	container = document.getElementById(DEFAULT_ALERT_CONTAINER_ID);
-
-	if (!container) {
-		container = document.createElement('div');
+	let alertFixed = document.getElementById(DEFAULT_ALERT_CONTAINER_ID);
 
-		container.id = DEFAULT_ALERT_CONTAINER_ID;
+	if (!alertFixed) {
+		alertFixed = buildFragment(TPL_ALERT_CONTAINER).querySelector(
+			'.alert-container.container'
+		);
 
-		document.body.appendChild(container);
+		alertFixed = document.body.appendChild(alertFixed);
 	}
 
+	// Creates a fragment for preventing React to unmount the alertContainer
+
+	container = document.createElement('div');
+	container.className = 'mb-3';
+
+	const fragmentContainer = document.querySelector(
+		`.alert-notifications.alert-notifications-fixed`
+	);
+
+	fragmentContainer.appendChild(container);
+
```

Comment:
About your question:

> I used metal-dom's [buildFragment](https://github.com/metal/metal.js/blob/master/packages/metal-dom/src/domNamed.js#L199) function for creating html from string, instead of declaratively and verbosely calling createElement etc calls.
> 
> Just for curiosity:
> 
> Should We avoid using `metal-dom` utilities?

I personally wouldn't use it for what you're doing here (creating two `div` elements); I don't think it's pulling its weight. Especially because you end up still doing manual DOM manipulation anyway afterwards (including another call to `createElement()`).

> If no, Do we have plans to migrate these utilities into our DXP codebase?

We haven't hard deprecated it yet. It's possible that the remaining useful bits of it might get migrated into DXP itself directly at some point. I personally haven't needed it at all so far, and I know @julien has been able to get by without it quite a bit. I've seen @markocikos use it in some PRs for things like event delegation which are a bit more cumbersome to implement from scratch.